scope.go

```diff
@@ -50,136 +50,117 @@ func NewScope() *Scope {
 // AddBreadcrumb adds new breadcrumb to the current scope
 // and optionaly throws the old one if limit is reached.
 func (scope *Scope) AddBreadcrumb(breadcrumb *Breadcrumb, limit int) {
+	scope.Lock()
```

Comment:
Why moving this one up?
The scope of the lock was actually smaller with the previous approach (no need to lock the timestamp work)